File: py/2b.py
def game_score(game):
    them = game[0]
    goal = game[1]
    A_offset = ord("A")
    X_offset = ord("X")
    them = ord(them) - A_offset
    goal = ord(goal) - X_offset # 0 lose 1 draw 2 win
    if goal == 1:
        return them + 1 + 3
    if goal == 2:
        return (them + 1) % 3 + 1 + 6
    return (them - 1) % 3 + 1

with open("2.input") as f:
    lines = [line.strip().split() for line in f.readlines()]

print(sum(map(game_score, lines)))

File: fsharp/2b.fs
let lines = System.IO.File.ReadLines("2.input");;

let toChar (x:string) =
    Seq.map char (x.Split ' ')

let scaleNearZero (x:int seq) =
    x 
    |> Seq.toList
    |> fun a -> [a[0] - int 'A'; a[1] - int 'X']

let toScore (x:int list) =
    let opponent = x[0]
    let goal = x[1]

    if goal = 1 then opponent + 1 + 3
    elif goal = 2 then (opponent + 1) % 3 + 1 + 6
    else (opponent + 2) % 3 + 1

let result = Seq.map (toChar >> Seq.map int >> scaleNearZero >> toScore) lines |> Seq.sum
